src/Dialog/DialogBottomSheet.tsx

```diff
@@ -0,0 +1,327 @@
+    if (!dialogRef.current) return
+    dialogRef.current.style.height = `${height}dvh`
+    setSnappedHeight(height)
+  }
+
+  // 🎪 EVENTS
+
+  const dragStop = (e: MouseEvent | TouchEvent) => {
+    if (!dialogRef.current) return
+
+    e.preventDefault()
+
+    isDragging.current = false
+    const sheetHeight = parseInt(dialogRef.current.style.height)
+
+    const isReduced = prefersReducedMotion()
+    dialogRef.current.style.transition = isReduced ? 'none' : '0.3s ease'
+
+    if (sheetHeight < 10) return hideBottomSheet('drag')
+    if (sheetHeight > 75) return updateSheetHeight(FULL_HEIGHT)
+
+    updateSheetHeight(HALF_HEIGHT)
+  }
+
+  const dragStart = (e: MouseEvent | TouchEvent) => {
+    if (!dialogRef.current) return
+
+    e.preventDefault()
+
+    if (e.type === 'touchstart' && 'touches' in e) {
+      startY.current = e.touches[0].pageY
+    } else if ('clientX' in e) {
+      startY.current = e.pageY
+    }
+
+    startHeight.current = parseInt(dialogRef.current.style.height)
+    isDragging.current = true
+    dialogRef.current.style.transition = 'none'
+  }
+
+  const dragging = (e: MouseEvent | TouchEvent) => {
+    if (!dialogRef.current || !isDragging.current) return
+
+    e.preventDefault()
+
+    let pageY
+    if (e.type === 'touchmove' && 'touches' in e) {
+      pageY = e.touches[0].pageY
+    } else if ('clientX' in e) {
+      pageY = e.pageY
+    }
+
+    const delta = startY.current - (pageY || 0)
+    const newHeight = startHeight.current + (delta / window.innerHeight) * 100
+    updateSheetHeight(newHeight)
+  }
+
+  const onSliderChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (!isDragging.current) {
+      const value = parseInt(e.target.value)
+      if (value === 1) {
+        updateSheetHeight(HALF_HEIGHT)
+      } else if (value === 2) {
+        updateSheetHeight(FULL_HEIGHT)
+      }
+    }
+  }
+
+  // 🪝 HOOKS
+
+  // Ensures an animation upon closing the component
+  useEffect(() => {
+    if (!fireDelayedOnClose) return
+    setIsOpen(false)
+    const timer = setTimeout(() => {
+      onClose(fireDelayedOnClose)
+    }, ANIMATION_DURATION)
+    return () => clearTimeout(timer)
+  }, [fireDelayedOnClose, onClose])
+
+  // Animates the bottom sheet in on mount
+  useEffect(() => {
+    showBottomSheet()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const currentSliderValue = snappedHeight === HALF_HEIGHT ? 1 : 2
+
+  return (
+    <FullScreenContainer
+      open={open}
+      onMouseUp={dragStop}
+      onMouseMove={dragging}
+      onTouchEnd={dragStop}
+      onTouchMove={dragging}
+    >
+      <Overlay onClick={() => hideBottomSheet('overlay')}></Overlay>
+      <Content
+        ref={dialogRef}
+        role={role}
+        open={open}
+        aria-labelledby={ariaLabelledby}
+        aria-describedby={ariaDescribedby}
+        aria-modal={ariaModal}
+        sx={sx}
+      >
+        <Box sx={{position: 'relative'}}>
+          <DraggableRegion
```

Comment:
I think it might be a little nicer if the close button was the first focusable element in the DOM. This way, arrow buttons don't move the sheet around until I explicitly focus the drag handle.

Reply:
Fixed.